Simplenote/src/androidTest/java/com/automattic/simplenote/Login.java

```diff
@@ -0,0 +1,43 @@
+package com.automattic.simplenote;
+
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+
+import static com.automattic.simplenote.utils.EspressoUITests.logOut;
+import static com.automattic.simplenote.utils.EspressoUITests.loginWithCredentials;
+import static com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertContains;
+import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;
+import static com.schibsted.spain.barista.interaction.BaristaEditTextInteractions.writeTo;
+import static java.lang.Thread.sleep;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class Login {
+
+    @Rule
+    public ActivityTestRule<NotesActivity> mActivityTestRule = new ActivityTestRule<>(NotesActivity.class);
+
+    @Test
+    public void loginWithCorrectCredentialsLogout() throws InterruptedException {
+        loginWithCredentials();
+        logOut();
+    }
+
+    @Test
+    public void loginWithInvalidCredentials() throws InterruptedException {
+        clickOn(R.id.button_login);
+        clickOn(R.id.button_email);
+        writeTo(R.id.input_email, "test.espresso.00005@gmail.com");
+        writeTo(R.id.input_password, "*-Re7]J4Ux8q)g?X");
+        clickOn(R.id.button);
+        sleep(2000);
```

Comment:
I noticed this is almost the same set of steps used in `loginWithCredentials()`. What do you think of writing a method that can handle both valid and invalid login credentials? That way if the login flow changes, there are fewer places to update the tests.

Reply:
improved:


before:
<img width="1018" alt="Screenshot 2020-05-27 at 17 33 58" src="https://user-images.githubusercontent.com/29093946/83047339-2528bf80-a040-11ea-94b2-eac848d60444.png">

after:
<img width="981" alt="Screenshot 2020-05-27 at 17 33 01" src="https://user-images.githubusercontent.com/29093946/83047333-222dcf00-a040-11ea-9612-4369ebe144a5.png">

added a test only inserting the wrong credentials - `loginWithInvalidCredentials()` on `EspressoUITests.java`, that way we can also do more combinations such as try to login with wrong credentials and then use the correct ones and others

Reply:
Thanks for that side by side comparison! What about writing a single `login()` method that takes the credentials as parameters, so you only need one method to try different combinations?

Reply:
updated, thanks! 😄